client/my-sites/customer-home/components/celebrate-launch-modal.jsx

```diff
@@ -18,23 +20,44 @@ function CelebrateLaunchModal( { setModalIsOpen, site, allDomains } ) {
 	const isBilledMonthly = site?.plan?.product_slug?.includes( 'monthly' );
 
 	const transformedDomains = allDomains.map( createSiteDomainObject );
+
+	const [ isInitiallyLoaded, setIsInitiallyLoaded ] = useState( false );
 	const [ clipboardCopied, setClipboardCopied ] = useState( false );
 	const clipboardButtonEl = useRef( null );
 	const hasCustomDomain = Boolean(
 		transformedDomains.find( ( domain ) => ! domain.isWPCOMDomain )
 	);
 
 	useEffect( () => {
-		// remove the celebrateLaunch URL param without reloading the page as soon as the modal loads
-		// make sure the modal is shown only once
+		dispatch( fetchSitePurchases( site?.ID ) );
+	}, [ dispatch, site?.ID ] );
+
+	const isLoading = useSelector( isFetchingSitePurchases );
+	const purchases = useSelector( ( state ) => getSitePurchases( state, site?.ID ) );
+
+	useEffect( () => {
+		if ( ! isLoading && ( ! isPaidPlan || purchases.length > 0 ) && ! isInitiallyLoaded ) {
+			setIsInitiallyLoaded( true );
+		}
+	}, [ isLoading, purchases, isInitiallyLoaded ] );
```

Comment:
I think we might be missing `isPaidPlan` in the dependency array:

```suggestion
	}, [ isLoading, isPaidPlan, purchases, isInitiallyLoaded ] );
```

Reply:
Very good point. Putting it in.